core/src/main/kotlin/com/ritense/valtimo/importchangelog/ImportChangelogDeploymentService.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ *  Licensed under EUPL, Version 1.2 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.ritense.valtimo.importchangelog
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.ritense.valtimo.contract.importchangelog.ChangesetDeployer
+import com.ritense.valtimo.importchangelog.domain.ImportChangeset
+import com.ritense.valtimo.importchangelog.repository.ImportChangesetRepository
+import liquibase.util.MD5Util
+import liquibase.util.StringUtil
+import mu.KotlinLogging
+import org.springframework.boot.context.event.ApplicationReadyEvent
+import org.springframework.context.event.EventListener
+import org.springframework.core.io.Resource
+import org.springframework.core.io.ResourceLoader
+import org.springframework.core.io.support.ResourcePatternUtils
+import org.springframework.transaction.annotation.Transactional
+import java.io.IOException
+import java.text.Normalizer
+import java.time.Instant
+import kotlin.jvm.optionals.getOrNull
+
+@Transactional
+class ImportChangelogDeploymentService(
+    private val resourceLoader: ResourceLoader,
+    private val importChangesetRepository: ImportChangesetRepository,
+    private val objectMapper: ObjectMapper,
+    private val changesetDeployers: List<ChangesetDeployer>,
+) {
+
+    @EventListener(ApplicationReadyEvent::class)
+    fun deployAll() {
```

Comment:
When do tables get cleared? I looked around a bit and couldn't find it. As per the story:

> Application property that deletes the contents of the checksum table and the permissions table when set to true. Set to false by default.

Reply:
But is we do that, how can we support multiple permission files?
right now we have: `document.permission.json` and `note.permission.json`

Reply:
This is why I think this should be done by the individual deployers. Have them look at the property, and if it's set then we delete the tables that will be imported (and the relevant records from the changeset table). So in this case, it's about permissions, which means the permissions table gets cleared first. After that, all .permission.json files get imported.